sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -31,6 +35,25 @@ public HttpMessage(Request request, ResponseClassifier responseClassifier)
             _propertyBag = new ArrayBackedPropertyBag<ulong, object>();
         }
 
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <exception cref="ArgumentException"></exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public HttpMessage(RestMessage message)
```

Comment:
This adds a constructor to HttpMessage to make it its own adapter to a RestMessage.  I don't like that it goes on the public surface of HttpMessage, so if we can handle it differently moving forward, that would be better.  It is also not clear that this works for the general case, i.e. would the pipeline function correctly if someone subclassed RestMessage and sent it with the Rest pipeline?  I don't think it works, but I haven't worked through it e2e.